src/witan/cic/episodes/transformation_report.clj

```diff
@@ -13,52 +13,90 @@
+      (into fields [(user-friendly-command-string (::wce/command edit)) (::wce/reason edit)]))))
+
+(def format-rows-xf
+  (comp
+   (mapcat (fn [[_ {::wce/keys [ssda903-episodes]}]] ssda903-episodes))
+   (x/sort-by (juxt ::wce/id ::wce/report-date ::wce/report-year))
+   (map format-row)))
 
 (defn ->excel [out-file episodes]
-  (xl/save-workbook!
-   out-file
-   (xl/create-workbook
-    "Transformation Report"
-    (x/into [["ID" "dob" "sex"
-              "report year"
-              ;; "report date" "ceased" "placement" "legal status" "care status"
-              "DECOM" "DEC" "PLACE" "LS" "CIN"
-              "file name" "sheet name" "row number"
-              "edit status" "reason"
-              "affecting report year"
-              ;; "new report date" "new ceased" "new placement" "new legal status" "new cares status"
-              "DECOM" "DEC" "PLACE" "LS" "CIN"
-              "affecting file name" "affecting sheet name" "affecting row number"
-              "original fields"
-              "DECOM" "DEC" "PLACE" "LS" "CIN"]]
-            (comp
-             (mapcat (fn [[_ {::wce/keys [ssda903-episodes]}]] ssda903-episodes))
-             (x/sort-by (juxt ::wce/id ::wce/report-date ::wce/report-year))
-             (map (fn [episode]
-                    (let [fields ((juxt ::wce/id (fn [e] (str (::wce/dob e))) ::wce/sex ::wce/report-year (fn [e] (str (::wce/report-date e))) (fn [e] (str (::wce/ceased e))) ::wce/placement ::wce/legal-status ::wce/care-status ::wcie/file-name ::wcie/sheet-name ::wcie/row-index)
-                                  episode)
-                          edit (or
-                                (some #(when (= (::wce/command %) :remove) %) (::wce/edit episode))
-                                (first (::wce/edit episode)))]
-                      (cond
-                        (::wce/new edit)
-                        (let [new (::wce/new edit)
-                              previous (::wce/previous edit)]
-                          (into fields
-                                [(user-friendly-command-string (::wce/command edit)) (::wce/reason edit)
-                                 (::wce/report-year new)
-                                 (str (::wce/report-date new)) (str (::wce/ceased new)) (::wce/placement new) (::wce/legal-status new) (::wce/care-status new)
-                                 (::wcie/file-name new) (::wcie/sheet-name new) (::wcie/row-index new)
-                                 "Original" (str (::wce/report-date previous)) (str (::wce/ceased previous)) (::wce/placement previous) (::wce/legal-status previous) (::wce/care-status previous)]))
+  (let [template (xl/load-workbook "template-transformation-report.xlsx")
+        sheet (xl/select-sheet "Transformation Report" template)]
+    (xl/add-rows! sheet
+                  (x/into []
+                          format-rows-xf
+                          episodes))
+    (xl/save-workbook! out-file template))
+  #_(xl/save-workbook!
```

Comment:
?

Reply:
that is the old save workbook function and the new one. I'm not sure which I want to go with atm. I'd like to be able to template the workbook, but I'm not able to make that work atm.